pygit2/index.py

```diff
@@ -353,6 +353,25 @@ def hex(self):
         """The id of the referenced object as a hex string"""
         return self.id.hex
 
+    def __str__(self):
+        return "<path={} id={} mode={}>".format(self.path, self.hex, self.mode)
+
+    def __repr__(self):
+        t = type(self)
+        return "<{}.{} path={} id={} mode={}>".format(
+            t.__module__, t.__qualname__, self.path, self.hex, self.mode
+        )
+
+    def __eq__(self, other):
+        if self is other:
+            return True
+        if not isinstance(other, IndexEntry):
+            return NotImplemented
+        return self.path == other.path and self.id == other.id and self.mode == other.mode
+
+    def __ne__(self, other):
```

Comment:
https://docs.python.org/3/reference/datamodel.html?highlight=__eq__#object.__eq__ says `__ne__` is the inverse of `__eq__` by default, so I don’t think this is needed?

Reply:
good catch

Reply:
True - I just read the README and pygit2 requires python3.6 where this is true (in python2 it is not true).